Update readmes and simplify repo (#53)
WayScience/NF1_SchwannCell_data_analysis: 1.train_models/README.md

# Develop the NF1 model
In this module, we extract patterns to discern between cells with WT and Null genotypes using cell morphologies from plates 3, 3 prime, and 5.

> After substantial development we have developed and optimized our final NF1 model with the same architecure as a logistic regression.

Reviewer: ```suggestion
After substantial development, we have developed and optimized our final NF1 model with the same architecture as a logistic regression.
```
Quick question on this statement:
> "same architecture as a logistic regression"

When you say this, it sounds like we made a whole new ML model. Remind me, do you use the sklearn package to create a logistic regression or was there another method?

Author: This is just another way of saying I used a logistic regression in a more formal way. Also, it is an sklearn model

> The training and testing sets are choose to create a 90-10 train-test split between wells, while approximately retaining a uniform genotype class distribution of cells within each data split.

Reviewer: ```suggestion
The training and testing sets were choosen to create a 90-10 percent train-test split between wells, while approximately retaining a uniform genotype class distribution of cells within each data split.
```

Reviewer: Recommend staying consistent with past or present tense. Also might not need to include percent, so feel free to add or ignore.

Author: I was trying to create an active voice, but your suggestion looks better

> The training and testing sets were choosen to create a 90-10 percent train-test split between wells, while approximately retaining a uniform genotype class distribution of cells within each data split.

Reviewer: ```suggestion
The training and testing sets were manually chosen to create a 90-10 percent train-test split between wells while approximately retaining a uniform genotype class distribution of cells within each data split.
```

Reviewer: Is this accurate to use manual here? If not, recommend clarifying what was manually done.

Author: I think so, this is opposed to a more purely algorithmic approach. I just thought of a better way to phrase this, which will be applied shortly

This minimizes the loss of cells when downsampling to the minority genotype class in each data split.

> We optimize the choosen model by performing a random search with stratified k-fold cross validation using the class-balanced training set.

Reviewer: ```suggestion
We optimize the chosen model by performing a random search with stratified k-fold cross-validation using the class-balanced training set.
```

> After optimization we retrain the NF1 model with the optimal regularization strength using the entire dataset, but excluding the test set.

Reviewer: ```suggestion
After optimization, we retrain the NF1 model with the optimal regularization strength using the entire dataset, but excluding the test set.
```

## Saving the model and results

> During optimization of the NF1 model we save the predicted genotype probabilities, true genotype labels, and the predicted genotype labels of the training, validation, and testing datasets for later evaluation.

Reviewer: ```suggestion
During optimization of the NF1 model, we save the predicted genotype probabilities, true genotype labels, and the predicted genotype labels of the training, validation, and testing datasets for later evaluation.
```
